mongodbatlas/resource_mongodbatlas_online_archive.go

```diff
@@ -98,6 +98,20 @@ func getMongoDBAtlasOnlineArchiveSchema() map[string]*schema.Schema {
 				},
 			},
 		},
+		"data_expiration_rule": {
+			Type:     schema.TypeList,
+			MinItems: 1,
+			MaxItems: 1,
+			Optional: true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"expire_after_days": {
+						Type:     schema.TypeInt,
+						Required: true,
```

Comment:
Currently defining this inner attribute as required as it is the only attribute in the inner object, and if object is left empty Atlas API responds with a 500 server error. In the future if the `data_expiration_rule` defines new attributes we can transition to optional.